diff --git a/node_modules/react-native-ui-lib/lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java b/node_modules/react-native-ui-lib/lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java
index 95c5e8e..c9d1b9a 100644
--- a/node_modules/react-native-ui-lib/lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java
+++ b/node_modules/react-native-ui-lib/lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java
@@ -9,7 +9,16 @@ public class RuntimeUtils {
     private static final Runnable sUIUpdateClosure = new Runnable() {
         @Override
         public void run() {
-            ReactContextHolder.getContext().getNativeModule(UIManagerModule.class).onBatchComplete();
+            try {
+              UIManagerModule uiManager = ReactContextHolder.getContext().getNativeModule(UIManagerModule.class);
+              if (uiManager != null) {
+                  uiManager.onBatchComplete();
+              } else {
+                  System.err.println("UIManagerModule is null, cannot call onBatchComplete");
+              }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
         }
     };
 
